modules/nextflow/src/test/groovy/FunctionalTests.groovy

```diff
@@ -750,4 +750,109 @@ class FunctionalTests extends Dsl2Spec {
         and:
         runner.session.fault.report ==~ /(?s).*-- Check script '(.*?)' at line: 10.*/
     }
+
+    def 'should handle dynamic cache directives' () {
+        given:
+        String config = '''
+            process {
+                withName: MyProcess {
+                    cache = { task.tag != 'mytag' }
+                }
+            }
+            '''
```

Comment:
This currently doesn't work because `task` is `null` when this closure gets evaluated. If you define a closure like this in a `nextflow.config` file, will it ever work? Isn't it attempting to capture a local variable that never exists in the context of the config file? In other words, `task`s come from scripts, not config files. If closures work the way I understand them to, referencing the `task` variable in `nextflow.config` won't work because tasks don't get defined until the processor executes a script.